Sources/Paywalls/Components/PaywallComponentLocalization.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallComponentLocalization.swift
+//
+//  Created by James Borthwick on 2024-09-03.
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public typealias LocaleID = String
+public typealias LocalizationDictionary = [String: String]
+public typealias LocalizationKey = String
```

Comment:
Worried about having non-namespaced typealiases here 🤔 Anybody who does `import RevenueCatUI` will get these typealises and could be general enough where they clash 😬 

Not sure if we have an opinions on these (even though I do love typeliases 😇) 

Would like some thoughts from @aboedo and other @RevenueCat/coresdk people on this 🙏

Reply:
These are defined in the RevenueCat module. If there is a collision, they can be resolved, but it would be a bit annoying. A few options:

1. Use weirder names
2. Prefix `RC` onto the type names, Obj-C style
3. Put these in an empty enum (with an unusual name) purely for name spacing purposes.

I'd opt for option three above, so for example:
```swift
public enum PaywallComponentType {
    public typealias LocaleID = String
}
```
Would then be accessed as
```swift
let foo: PaywallComponentType.TestSymbol = "Hello"
```
or to make it less verbose, you could choose to type alias the longer type:
```swift
// assumes no collision
public typealias TestSymbol = PaywallComponentType.TestSymbol
let foo: TestSymbol = "Hello"

// if there were a collision
public typealias RCTestSymbol = PaywallComponentType.TestSymbol
let foo: RCTestSymbol = "Hello"
```

Reply:
@jamesrb1 I like putting the typealias in an enum if we are able to do that!

Reply:
I just edited my response with another idea, to re-used a related public struct. Thoughts? @joshdholtz

Reply:
Putting in `PaywallComponentsData` sounds good to me 👍

Reply:
> Put these in the PaywallComponentsData struct (which is public).

FWIW I think this is the way that SK2 has been handling it

Reply:
cc @MarkVillacampa thoughts?

Reply:
I [moved them](3bc2c7d0bcfde2938389d652874c42da538eb31f) into the `PaywallComponent` struct, which seemed like the most "generic" place to put them for namespacing purposes.

_Aside: Swift needs an equivalent to C#'s "using" statement._